src/update-process-env.js

```diff
@@ -34,7 +31,7 @@ function updateProcessEnv (launchEnv) {
     }
 
     for (let key in envToAssign) {
-      if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key)) {
+      if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key) || (!process.env[key] && envToAssign[key])) {
```

Comment:
Right, that explains the `!process.env[key]` part of the expression but not the `&& envToAssign[key]`. It's minor though. I'll just merge anyway.